### ISSUE TITLE
Could not find plugin "@typescript-eslint"

### ISSUE THREAD

**Issue by Javid-Izadfar** [antfu/nuxt-module-eslint-config#1]
### Describe the bug

even on a brand new project, running `eslint .` causes error:

`TypeError: Key "rules": Key "@typescript-eslint/no-unused-vars": Could not find plugin "@typescript-eslint".`

### Reproduction

https://github.com/Javid-Izadfar/nuxt3-eslint-module-test

### System Info

```Shell
System:
    OS: Linux 5.15 Ubuntu 20.04 LTS (Focal Fossa)
    CPU: (12) x64 Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz
    Memory: 766.12 MB / 7.71 GB
    Container: Yes
    Shell: 5.0.16 - /bin/bash
  Binaries:
    Node: 18.15.0 - ~/.nvm/versions/node/v18.15.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v18.15.0/bin/yarn
    npm: 9.6.3 - ~/.nvm/versions/node/v18.15.0/bin/npm
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by RihanArfan** [comment]
I can replicate this both via CLI and VSCode extension output error. 

```
$ pnpm eslint .

Oops! Something went wrong! :(

ESLint: 8.56.0

TypeError: Key "rules": Key "@typescript-eslint/no-unused-vars": Could not find plugin "@typescript-eslint".
    at throwRuleNotFoundError (D:\projects\bereal-web\node_modules\.pnpm\eslint@8.56.0\node_modules\eslint\lib\config\rule-validator.js:66:11)
    at RuleValidator.validate (D:\projects\bereal-web\node_modules\.pnpm\eslint@8.56.0\node_modules\eslint\lib\config\rule-validator.js:128:17)
    at [finalizeConfig] (D:\projects\bereal-web\node_modules\.pnpm\eslint@8.56.0\node_modules\eslint\lib\config\flat-config-array.js:231:23)
    at FlatConfigArray.getConfig (D:\projects\bereal-web\node_modules\.pnpm\@humanwhocodes+config-array@0.11.14\node_modules\@humanwhocodes\config-array\api.js:938:55)
    at FlatConfigArray.isFileIgnored (D:\projects\bereal-web\node_modules\.pnpm\@humanwhocodes+config-array@0.11.14\node_modules\@humanwhocodes\config-array\api.js:962:15)
    at D:\projects\bereal-web\node_modules\.pnpm\eslint@8.56.0\node_modules\eslint\lib\eslint\eslint-helpers.js:312:49
    at Array.reduce (<anonymous>)
    at entryFilter (D:\projects\bereal-web\node_modules\.pnpm\eslint@8.56.0\node_modules\eslint\lib\eslint\eslint-helpers.js:299:28)
    at Object.isAppliedFilter (D:\projects\bereal-web\node_modules\.pnpm\@nodelib+fs.walk@1.2.8\node_modules\@nodelib\fs.walk\out\readers\common.js:12:31)
    at AsyncReader._handleEntry (D:\projects\bereal-web\node_modules\.pnpm\@nodelib+fs.walk@1.2.8\node_modules\@nodelib\fs.walk\out\readers\async.js:86:20)
```

<details><summary>Details</summary>
<p>

### pnpm nuxi info
```
------------------------------
- Operating System: Windows_NT
- Node Version:     v20.11.0
- Nuxt Version:     3.9.2
- CLI Version:      3.10.0
- Nitro Version:    2.8.1
- Package Manager:  pnpm@8.14.1
- Builder:          -
- User Config:      modules, googleFonts, pwa, image, ssr, runtimeConfig, experimental, future, devtools
- Runtime Modules:  @nuxtjs/tailwindcss@6.10.4, shadcn-nuxt@0.8.7, nuxt-icon@0.6.8, @nuxt/image@1.2.0, @vueuse/nuxt@10.7.2, @vueuse/motion/nuxt@2.0.0, @vite-pwa/nuxt@0.4.0, @nuxtjs/google-fonts@3.1.3, @nuxtjs/fontaine@0.4.1, nuxt-module-eslint-config@0.0.1
- Build Modules:    -
------------------------------
```

### eslint.config.js
```js
import NuxtEslintConfig from "./.nuxt/eslint.config.mjs";
import eslintConfigPrettier from "eslint-config-prettier";

export default [...NuxtEslintConfig, eslintConfigPrettier];
```

### snippet of package.json
```
    "eslint": "^8.56.0",
    "eslint-config-prettier": "^9.1.0",
    "nuxt": "^3.9.2",
    "nuxt-module-eslint-config": "^0.0.1",
```

### pnpm version
`v8.14.1`

</p>
</details> 

I can see the playground has some differences in packages. I've tried installing them to see, but see no difference. I've also tried using pnpm overrides on @typescript-eslint/eslint-plugin with both v6 and v7 pre releases and still getting the error.

**Comment by antfu** [comment]
It should be fixed in v0.0.2. Thanks.
